lucene/core/src/java/org/apache/lucene/search/LeafFieldComparator.java

```diff
@@ -116,4 +116,7 @@
    * obtain the current hit's score, if necessary. */
   void setScorer(Scorable scorer) throws IOException;
 
+  /** Publishes feature values for the given docID
+   */
+  T leafValue(int doc) throws IOException;
```

Comment:
Maybe rename to `getDocValue`?  Can you expand the javadocs to state that the provided `doc` must only go forward?